### PR TITLE
Update/fbfw 146 arrow logic unit to arrow

### PR DESCRIPTION
**Pull request by bryancodazen**
This PR adds arrow logic to connect an arrow to another arrow. To draw an arrow to another arrow, use the drawArrow function and pass in the `receiverType` arg as "arrow" and the `receiverIdentifier` as the orderID which drew that arrow. 

Screen Shot 2022-04-29 at 4 57 15 PM

### CODE REVIEW

File: beta-src/src/utils/map/drawArrow.ts [hunk 1]
```diff
@@ -22,190 +22,208 @@ export default function drawArrow(
...
+
+        if (toTerritoryEl && receiverUnit) {
+          receiverUnit = receiverUnit.parentNode;
+
+          const {
+            x: rUnitX,
+            y: rUnitY,
+            width: rUnitW,
+            height: rUnitH,
+          } = receiverUnit.getBBox();
```
Comment (mariomelchor) [inline beta-src/src/utils/map/drawArrow.ts:78]: Nit: Can we inline this like we did above.

**Review by mariomelchor — COMMENTED** [review]
For maintainability this file is quite hard to follow with all the if elseif checks.Anything we can do to clean this file up?

File: beta-src/src/utils/map/drawArrow.ts (hunk 1 above)
Comment (bryancodazen) [reply beta-src/src/utils/map/drawArrow.ts:78]: We can't inline this as prettier automatically formats it this way.

**Comment by bryancodazen** [comment]
> For maintainability this file is quite hard to follow with all the if elseif checks.Anything we can do to clean this file up?

The if, else logic is necessary to know where to dispatch/receive an arrow pointer from. Originally, the file had if/else statements in every switch statement but I improved the file by putting them all after the switch statement.

**Comment by bryancodazen** [comment]
> For maintainability this file is quite hard to follow with all the if elseif checks.Anything we can do to clean this file up?

I can see about putting the if/else checks in a separate function to improve readability of this file.

**Comment by bryancodazen** [comment]
> For maintainability this file is quite hard to follow with all the if elseif checks.Anything we can do to clean this file up?

I made changes to this file on https://github.com/codazen/webDiplomacy/pull/102 so I will make improvements there.

**Review by justinyoocodazen — APPROVED** [review]

**Review by mariomelchor — APPROVED** [review]
LGTM
